Hackaton 2024: resolved some doc issues (#2574)
openturns/openturns: python/doc/theory/probabilistic_modeling/copulas.rst

Copulas
-------

Let :math:`F` be a multivariate distribution function of dimension
:math:`\inputDim` whose marginal distribution functions are
:math:`F_1,\dots,F_\inputDim`. There exists a copula
:math:`C: [0,1]^\inputDim \mapsto [0,1]` of dimension :math:`\inputDim` such that for :math:`\vect{x}\in \Rset^\inputDim`, we have:

.. math::

   \begin{aligned}
       F(\vect{x})  = C \left( F_1(x_1),\cdots,F_d(x_d) \right)
    \end{aligned}

where :math:`F_i` is the cumulative distribution function of the margin  :math:`X_i`.

> In the case of continuous marginal distributions, for all :math:`\vect{u}\in[0,1]^d`, the copula is uniquely defined by:

Reviewer: Please keep each line under 79 characters.
```suggestion
In the case of continuous marginal distributions, for all
:math:`\vect{u}\in[0,1]^d`, the copula is uniquely defined by:
```

Copulas
-------

Let :math:`F` be a multivariate distribution function of dimension

> :math:`d` whose marginal distribution functions are
> :math:`F_1,\dots,F_d`. There exists a copula
> :math:`C: [0,1]^d \mapsto [0,1]` of dimension :math:`d` such that for :math:`\vect{x}\in \Rset^d`, we have:

Reviewer: ```suggestion
:math:`\inputDim` whose marginal distribution functions are
:math:`F_1,\dots,F_\inputDim`. There exists a copula
:math:`C: [0,1]^\inputDim \mapsto [0,1]` of dimension :math:`\inputDim` such that for :math:`\vect{x}\in \Rset^\inputDim`, we have:
```

Author: In that PR, I introduced the math_notations in the section "Reliability and sentitivity". It would be too much to do all the other files at once. I propose to accept this PR and then to continue section by section.
Do you agree?

Reviewer 2: I don't see the problem with accepting this suggestion (and the others of this kind). All it takes is to click on the button...

Reviewer 2: And to change the others, you could do it quite quickly by searching in this file for the strings `_d`, `{d}`,  `^d` and 
```
`d`
```

Author: No I don't agree.
@joseph: if I accept the suggestion of Michael, some files in tha section will ba cieherent with math_nottaions and the otherones not. It will be confusing for the reader.
@mbaudin47 : math_notations do not concern only the input dimension ....

I really prefer that we do these modifications in a new PR that will follow this one very soon.

.. math::

   \begin{aligned}

>        F(\vect{x})  = C \left( F_1(x_1),\cdots,F_d(x_d) \right)

Author: idem

>        F(\vect{x})  = C \left( F_1(x_1),\cdots,F_d(x_d) \right)

Reviewer 2: ```suggestion
       F(\vect{x})  = C \left( F_1(x_1),\cdots,F_d(x_\inputDim) \right)
```
Reproposing @mbaudin47 's suggestion

    \end{aligned}

where :math:`F_i` is the cumulative distribution function of the margin  :math:`X_i`.

In the case of continuous marginal distributions, for all :math:`\vect{u}\in[0,1]^d`, the copula is uniquely defined by:

.. math::

   \begin{aligned}
   C(\vect{u}) & = F(F_1^{-1}(u_1),\hdots,F_d^{-1}(u_d))\\
          & = \Prob{U_1 \leq u_1, \hdots, U_d \leq u_d}
    \end{aligned}

> where :math:`U_i = F_i(X_i)` is the random variate following the uniform distribution on :math:`[0,1]`.

Reviewer 2: ```suggestion
where :math:`U_i = F_i(X_i)` is a random variable following the uniform distribution on :math:`[0,1]`.
```

> where :math:`U_i = F_i(X_i)` is a random variable following the uniform distribution on :math:`[0,1]`.
> 
> A copula of dimension :math:`d` is the restriction to the unit cube :math:`[0,1]^d` of a
> multivariate distribution function with uniform univariate marginals on :math:`[0,1]`.
> It has the following properties:
> 
> - :math:`\forall \vect{u},\vect{v}\in[0,1]^d, |C(\vect{u})-C(\vect{v})|\leq \sum_{i=1}^d |u_i-v_i|`,
> 
> - for all :math:`\vect{u}` with at least one component equal to 0, :math:`C(\vect{u})=0`,
> 
> - :math:`C` is :math:`d`-increasing which means that:
> 
>   .. math::
> 
>    \sum_{i_1=1}^2 \dots \sum_{i_d=1}^2 (-1)^{i_1 + \dots + i_d} C(x_{1i_1}, \dots, x_{ni_d})\geq 0
> 
> 
> where :math:`x_{j1}=a_j` and :math:`x_{j2}=b_j` for all :math:`j \in \{1,\dots,d\}` and
> :math:`\vect{a}`, :math:`\vect{b}\in[0,1]^d`, :math:`\vect{a}\leq \vect{b}`,
> 
> -  :math:`\vect{u}` with all its components equal to 1 except  :math:`u_k`,
>    :math:`C(\vect{u})=u_k`.
> 

Reviewer 2: ```suggestion
where :math:`U_i = F_i(X_i)` is a random variable following the uniform distribution on :math:`[0,1]`.

A copula of dimension :math:`\inputDim` is the restriction to the unit cube :math:`[0,1]^\inputDim` of a
multivariate distribution function with uniform univariate marginals on :math:`[0,1]`.
It has the following properties:

- :math:`\forall \vect{u},\vect{v}\in[0,1]^\inputDim, |C(\vect{u})-C(\vect{v})|\leq \sum_{i=1}^\inputDim |u_i-v_i|`,

- for all :math:`\vect{u}` with at least one component equal to 0, :math:`C(\vect{u})=0`,

- :math:`C` is :math:`\inputDim`-increasing which means that:

  .. math::

   \sum_{i_1=1}^2 \dots \sum_{i_\inputDim=1}^2 (-1)^{i_1 + \dots + i_\inputDim} C(x_{1i_1}, \dots, x_{ni_\inputDim})\geq 0

where :math:`x_{j1}=a_j` and :math:`x_{j2}=b_j` for all :math:`j \in \{1,\dots,\inputDim\}` and
:math:`\vect{a}`, :math:`\vect{b}\in[0,1]^\inputDim`, :math:`\vect{a}\leq \vect{b}`,

-  :math:`\vect{u}` with all its components equal to 1 except  :math:`u_k`,
   :math:`C(\vect{u})=u_k`.
```

> The copula represents the part of the joined cumulative density function which is not described by the

Reviewer 2: ```suggestion
The copula represents the part of the joint cumulative density function which is not described by the
```

marginal distributions. It models the dependence structure of the input variables.

> In the probabilistic modelisation issue, the choice of the dependence structure is disconnected from the
> choice of the marginal distributions.

Reviewer 2: ```suggestion
Note that a multivariate distribution is characterized by its marginal distributions and its copula.
Therefore, a multivariate distribution can be built by choosing the marginals and the copula independently.
```
